api/pkg/backend/service.go

```diff
@@ -205,17 +204,60 @@ func (s *APIService) CreateBackend(request *restful.Request, response *restful.R
 	actx := request.Attribute(c.KContext).(*c.Context)
 	backendDetail.TenantId = actx.TenantId
 	backendDetail.UserId = actx.UserId
+
+	storageTypes, err := s.listStorageType(ctx, request, response)
+	if err != nil {
+		log.Errorf("failed to list backend storage type: %v\n", err)
+		response.WriteError(http.StatusInternalServerError, err)
+		return
+	}
+	foundType := typeExists(storageTypes.Types, backendDetail.Type)
+	if !foundType {
+		log.Errorf("failed to retrieve backend type: %v\n", err)
+		response.WriteError(http.StatusBadRequest, err)
+		return
+	}
+
+	backendDetailtry := &s3.BackendDetailS3{}
+	backendDetailtry.Id = backendDetail.Id
+	backendDetailtry.Name = backendDetail.Name
+	backendDetailtry.Type = backendDetail.Type
+	backendDetailtry.Region = backendDetail.Region
+	backendDetailtry.Endpoint = backendDetail.Endpoint
+	backendDetailtry.BucketName = backendDetail.BucketName
+	backendDetailtry.Access = backendDetail.Access
+	backendDetailtry.Security = backendDetail.Security
+
+	_, err = s.s3Client.BackendCheck(ctx, backendDetailtry)
+	err1 := errors.New("failed to create backend due to invalid credentials")
+
+	if err != nil {
+		log.Errorf("failed to create backend due to wrong credentials: %v\n", err)
+		response.WriteError(http.StatusBadRequest, err1)
+		return
+	}
+
 	res, err := s.backendClient.CreateBackend(ctx, &backend.CreateBackendRequest{Backend: backendDetail})
 	if err != nil {
 		log.Errorf("failed to create backend: %v\n", err)
 		response.WriteError(http.StatusInternalServerError, err)
 		return
 	}
 
-	log.Info("Create backend successfully.")
+	log.Info("Created backend successfully.")
 	response.WriteEntity(res.Backend)
 }
 
+func typeExists(slice []*backend.TypeDetail, inputType string) bool {
```

Comment:
@vineela1999 suggest to rename variable slice (type vs. what the variable holds)

Reply:
code is already merged